src/main/scala/calc/Compiler.scala

```diff
@@ -13,6 +13,17 @@ object Compiler {
 
   private final val MainClassFullName = MainObjectFullName + "$"
 
+  import irt.BinaryOp._
+  private final val binaryOpMap = Map("+" -> Double_+, "-" -> Double_-,
+      "*" -> Double_*, "/" -> Double_/)
+
+  private final val nativeFun = Map(
+    "Ident(sin)" -> "sin__D__D",
+    "Ident(cos)" -> "cos__D__D",
+    "Ident(sqrt)" -> "sqrt__D__D",
+    "Ident(pow)" -> "pow__D__D__D"
```

Comment:
That's a code smell, indexing by the `toString` of trees! You should definitely only put `sin` as key.